apps/communications/contacts/test/marionette/details_test.js

```diff
@@ -110,4 +110,28 @@ marionette('Contacts > Details', function() {
     var menuOptions = sysMenu.findElements('button');
     assert.ok(menuOptions.length >= 2);
   });
+
+  test('Fav Contact in Reg/Fav List', function() {
+    //Set a Contact as a Favorite Contact
+    contactsData.createMozContact(testContact);
+    client.helper.waitForElement(selectors.listContactFirstText).click();
+    subject.waitSlideLeft('details');
+    client.helper.waitForElement(selectors.detailsFavoriteButton).click();
+    var nameNode = client.helper.waitForElement(selectors.detailsHeader);
+    client.waitFor(function() {
+      return nameNode.getAttribute('class').indexOf('favorite') != -1;
+    });
+    //Wait for return to Contacts List
+    subject.backToList();
+
+    //Verify Contact is in Regular Contacts List
+    var cName = client.findElement(selectors.listContactFirstText);
+    assert.ok(cName.displayed(), 'Element should exist.');
```

Comment:
As a rule of thumb: use custom messages [only when it adds more information to the error](https://wiki.mozilla.org/B2G/QA/Automation/Style_Guide/Best_Practices#Assertion_Messages).

Unlike the example given in the link, here we don't hide any data here. So it's not a real problem. Nevertheless, it's a good habit to have, and it helps in having a consistent code base.